pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/LanguageRuleMapping.java

```diff
@@ -0,0 +1,87 @@
+package sfdc.sfdx.scanner.pmd;
+
+import java.util.*;
+
+/**
+ * Maintains mapping between languages and their XML paths
+ */
+public class LanguageRuleMapping {
```

Comment:
What does this class provide that isn't achieved by keeping `categoryPathsByLanguage` and `rulesetPathsByLanguage` as properties of `PmdRuleCataloger`?

Reply:
It encapsulates the details of how we categorize XML files. We may choose to add more demarcations and Cataloger doesn't have to know about the logic we use for it. And it keeps the Cataloger class cohesive and its size under control.
(I need to rename this class to something more relevant)

Reply:
Just remember that it's possible to have too many layers of indirection. To me, this feels like it might be too many.